### PR TITLE
[1.2.0] Implement ExecuteOnKeys functionality [API-912]

### LINKED ISSUE #662: Implement Map ExecuteOnKeys method
`func (m *Map) ExecuteOnKeys(ctx context.Context, processor interface{}, key interface{}...) ([]interface{}, error)` is missing. We need to add it and the corresponding test.

### Pointers
* https://github.com/hazelcast/hazelcast-go-client/pull/626/files
* https://github.com/hazelcast/hazelcast-go-client/pull/630/files

### PR DESCRIPTION
**Pull request by utku-caglayan**
Resolves #662

### CODE REVIEW

File: internal/proto/codec/map_execute_on_key_codec.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ const (
 	MapExecuteOnKeyCodecRequestInitialFrameSize = MapExecuteOnKeyCodecRequestThreadIdOffset + proto.LongSizeInBytes
 )
 
-// Applies the user defined EntryProcessor to the entry mapped by the key. Returns the the object which is result of
+// Applies the user defined EntryProcessor to the entry mapped by the key. Returns the object which is result of
```
Comment (yuce) [inline internal/proto/codec/map_execute_on_key_codec.go:33]: No need to change the codecs. These are supposed to be auto-generated.

File: proxy_map.go [hunk 2]
```diff
@@ -908,20 +930,29 @@ func (m *Map) loadAll(ctx context.Context, replaceExisting bool, keys ...interfa
...
 }
 
+func (m *Map) convertToDataList(keys []interface{}) ([]*serialization.Data, error) {
+	keyDatas := make([]*serialization.Data, 0, len(keys))
+	for _, key := range keys {
+		// todo check if this null check done with validateAndSerialize cause problem for keys
+		if keyData, err := m.validateAndSerialize(key); err != nil {
+			return nil, err
+		} else {
+			keyDatas = append(keyDatas, keyData)
```
Comment (yuce) [inline proxy_map.go:950]: Can you change this to:
```
keyData, err := err := m.validateAndSerialize(key)
if err != nil  {
   return nil, err
}
keyDatas = append(keyDatas, keyData)
```

**Review by yuce — APPROVED** [review]
Looks good! Except two minor things.

File: proxy_map.go (hunk 2 above)
Comment (utku-caglayan) [reply proxy_map.go:950]: I will be fixing that, thanks for that catch. I wasn't sure to use `validateAndSerialize` with alternative being `convertToData` which is essentially same minus a nil check. 
I know that we support "null" as value on our map structure but do we support null keys? If not I validateAndSerialize seems fine here

Comment (yuce) [reply proxy_map.go]: No, we don't support `nil` for neither the key nor the value. `validateAndSerialize` is fine here. I think just reformatting that code is fine.

File: proxy_map.go [hunk 3]
```diff
@@ -931,20 +953,29 @@ func (m *Map) loadAll(ctx context.Context, replaceExisting bool, keys ...interfa
...
 		request = codec.EncodeMapLoadGivenKeysRequest(m.name, keyDatas, replaceExisting)
 	}
 	_, err := m.invokeOnRandomTarget(ctx, request, nil)
 	return err
 }
 
+func (m *Map) convertToDataList(keys []interface{}) ([]*serialization.Data, error) {
+	keyDatas := make([]*serialization.Data, 0, len(keys))
+	for _, key := range keys {
+		// todo check if this null check done with validateAndSerialize cause problem for keys
```
Comment (yuce) [inline proxy_map.go:969]: This shouldn't be a problem.

**Review by yuce — CHANGES_REQUESTED** [review]

File: proxy_map.go (hunk 3 above)
Comment (utku-caglayan) [reply proxy_map.go]: Thanks for checking that, removed

**Review by yuce — APPROVED** [review]
Thanks, looks good!

**Pull request merged**